ddtrace/tracer/spancontext.go

```diff
@@ -330,7 +330,17 @@ func (t *trace) finishedOne(s *span) {
 			atomic.AddUint64(&tr.droppedP0Spans, uint64(len(t.spans)))
 			atomic.AddUint64(&tr.droppedP0Traces, 1)
 		}
-		return
+		// if trace sampling decision is drop, we still want to send single spans
+		var singleSpans []*span
```

Comment:
Please find a way to not create this slice if all spans are kept, which is the case 95%+ of the time.

We also need to benchmark this `finishedOne` call (perhaps via `span.Finish`) to make sure we are not regressing for the common path (without any sampling rules).

Reply:
I am iterating over t.spans a line later